API Reference article (#10)
santiment/articles: src/api-reference/index.md

# API Reference

This page contains a reference of all the APIs provided by Santiment.

## Authentication

> Some of the APIs requite to have a valid API, belonging to an account with a

Reviewer: require*

# API Reference

This page contains a reference of all the APIs provided by Santiment.

## Authentication

> Some of the APIs require to have a valid API, belonging to an account with a

Reviewer 2: ```suggestion
Some of the APIs require to have a valid API key, belonging to an account with a
```

> paid subscription to access all the data. The API key can be generates on your

Reviewer 2: ```suggestion
paid subscription to access all the data. The API key can be generated on your
```

[Account Settings](https://app.santiment.net/account#api-keys) page.

After that you need to pass the API key as an additional HTTP header
`Authorization: Apikey <YOUR_OWN_API_KEY>`. An example how to do that using curl:

```bash
curl \
  -X POST \
  -H "Content-Type: application/graphql" \
  -H "Authorization: Apikey <YOUR_OWN_API_KEY>"\

>   --data '<YOUR_OWN_QUERY>}' \

Reviewer 2: ```suggestion
  --data '<YOUR_OWN_QUERY>' \
```

at the #support channel

## Live API Explorer

Santiment API uses [GraphQL](https://graphql.org). This is an alternative to REST,
which allows a lot of flexibility in the way the data is requested. You can request
exactly the data you need and also easily batch requests together.

There is a live explorer, where you can run queries directly from the browser. The
explorer is accessible here: [https://api.santiment.net/graphiql](https://api.santiment.net/graphiql)

Here is an example of running a query and seeing the results directly in the browser:

[https://api.santiment.net/graphiql?query=%7B%0A%20%20transactionVolume(slug%3A%22santiment%22%2C%20from%3A%222019-01-01T00%3A00%3A00Z%22%2C%20to%3A%222019-03-01T00%3A00%3A00Z%22)%20%7B%0A%20%20%20%20datetime%0A%20%20%20%20transactionVolume%0A%20%20%7D%0A%7D](https://api.santiment.net/graphiql?query=%7B%0A%20%20transactionVolume(slug%3A%22santiment%22%2C%20from%3A%222019-01-01T00%3A00%3A00Z%22%2C%20to%3A%222019-03-01T00%3A00%3A00Z%22)%20%7B%0A%20%20%20%20datetime%0A%20%20%20%20transactionVolume%0A%20%20%7D%0A%7D)

## Running requests with `curl`

> You can run requests with curl by passing the GraphQL query as the body of the request. Here is an exmaple how to run the transactionVolume query:

Reviewer 3: ```suggestion
You can run requests with curl by passing the GraphQL query as the body of the request. Here is an exmaple how to run the `transactionVolume` query:
```

```bash
curl \
  -X POST \
  -H "Content-Type: application/graphql" \
  --data '{ transactionVolume(slug:"santiment", from:"2019-01-01T00:00:00Z", to:"2019-03-01T00:00:00Z") { datetime transactionVolume }}' \
  https://api.santiment.net/graphql
```

## Glossary

> There some terms used in this document. Here is a list and discription of them:

Reviewer 2: ```suggestion
There are some terms used in this document. Here is a list and discription of them:
```

* `slug` - this is an ID we use in the API to identify projects. You can find the slug
of the projects by using the `allProjects` API
* `ISO 8601 format` - this is the date time format used in the API. The format is `<year>-<month>-<day>T<hour>:<minute>:<seconds>Z`. For example Jan 10th 2019 12:34:56 is `2019-01-10T12:34:56Z`

> * `API key` - this is your API for accessing the premium features in the API. See the `Authentication` section for more details

Reviewer 2: ```suggestion
* `API key` - this is your API key for accessing the premium features in the API. See the `Authentication` section for more details
```

## Available APIs

### Transaction volume

**Tier: Free**

**Max lag: 3 block confirmation**

**Max resolution: 5min**

Tracks the sum of all on-chain transactions done during a day for a given asset.

Parameters:

| Parameter | Description                                                            |
| --------- | ---------------------------------------------------------------------- |
| slug      | The slug of the project                                                |
| from      | From which date to return the volume in ISO 8601 format                |
| to        | Till which date to return the volume in ISO 8601 format                |
| interval  | The intervals that should be returned. Default is `1d`, which is daily |

The query looks like this:

> ```js

Reviewer 2: you gan use `graphql` instead of `js`

Author: There is no `graphql` support in pandoc renderer. JS was the closest I can find that colors the queries in some sane way.

    
    datetime
    transactionVolume
  }
}
```

**[Run in explorer](https://api.santiment.net/graphiql?query=%7B%0A%20%20transactionVolume(slug%3A%22santiment%22%2C%20from%3A%222019-01-01T00%3A00%3A00Z%22%2C%20to%3A%222019-03-01T00%3A00%3A00Z%22)%20%7B%0A%20%20%20%20datetime%0A%20%20%20%20transactionVolume%0A%20%20%7D%0A%7D)**

Supported assets:

| Bitcoin            | Ethereum           | ERC-20             | Ripple | EOS | EOS tokens |
| ------------------ | ------------------ | ------------------ | ------ | --- | ---------- |
| :white_check_mark: | :white_check_mark: | :white_check_mark: | WIP    | WIP | WIP        |

See also [getMetric](#the-getmetric-api) where you can find the same metric with support for more assets.

### Daily active addresses

**Tier: Free**

**Max lag: 1 day**

**Max resolution: daily**

Tracks the number of unique addresses that make transfers on-chain on a daily basis.

Parameters:

| Parameter | Description                                                            |
| --------- | ---------------------------------------------------------------------- |
| slug      | The slug of the project                                                |

> | from      | From which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| from      | From which date to return the active addresses in ISO 8601 format                |
```

> | to        | Till which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| to        | Till which date to return theactive addresses in ISO 8601 format                |
```

The `getMetric` API gives a unified API for many different metrics across the whole universe of supported assets.
The query allows to get metadata about a given metric and asset, as well as fetch the values of the metric in a
nice unified way.

Currently the metrics are updated 4 times a day. We are working on reducing this interval.

Supported assets:

| Bitcoin            | Ethereum           | ERC-20             | Ripple             | EOS                | EOS tokens         |
| ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ |
| :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |

#### timeseriesData

To fetch the values for a given metric, slug and time interval you can use the `timeseriesData` subquery of the `getMetrics` API.

Parameters:

| Parameter   | Description                                                            |
| ----------- | ---------------------------------------------------------------------- |
| slug        | The slug of the project                                                |

> | from        | From which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| from        | From which date to return the values in ISO 8601 format                |
```

> | to          | Till which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| to          | Till which date to return the values in ISO 8601 format                |
```

| interval    | The intervals that should be returned. Default is `1d`, which is daily |

> | aggregation | The aggregation to be used when fetching data for longer intervals     |

Reviewer 2: Maybe add: `If aggregation is not provided - every metric has a default one that fits best`

```js
{
  getMetric(metric:"transaction_volume") {
    metadata {
      minInterval
      defaultAggregation
    }
  }
}
```

**[Run in explorer](https://api.santiment.net/graphiql?query=%7B%0A%09getMetric(metric%3A%22transaction_volume%22)%20%7B%0A%20%20%20%20metadata%20%7B%0A%20%20%20%20%20%20minInterval%0A%20%20%20%20%20%20defaultAggregation%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A)**

### Github Activity

**Tier: free**

**Max lag: a few hours**

**Max resolution: 1min**

Returns the number of GitHub events generated by a given project repositories. Each project has one or more repositories/organizations that are tracked. A longer description of how the activity is being tracked can be found in this article: [Tracking GitHub activity of crypto projects — introducing a better approach](https://medium.com/santiment/tracking-github-activity-of-crypto-projects-introducing-a-better-approach-9fb1af3f1c32)

Parameters:

| Parameter | Description                                                            |
| --------- | ---------------------------------------------------------------------- |
| slug      | The slug of the project                                                |

> | from      | From which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| from      | From which date to return the activity in ISO 8601 format                |
```

> | to        | Till which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| to        | Till which date to return the activity in ISO 8601 format                |
```

    datetime
    activity
  }
}
```

**[Run in explorer](https://api.santiment.net/graphiql?query=%7B%0A%20%20githubActivity(slug%3A%22santiment%22%2C%20from%3A%222019-01-01T00%3A00%3A00Z%22%2C%20to%3A%222019-06-01T00%3A00%3A00Z%22%2C%20interval%3A%221d%22)%20%7B%0A%20%20%20%20datetime%0A%20%20%20%20activity%0A%20%20%7D%0A%7D)**

### Dev Activity

**Tier: free**

**Max lag: a few hours**

**Max resolution: 1min**

Returns the number of GitHub events generated by a given project repositories that are connected with development actions. Each project has one or more repositories/organizations that are tracked. A longer description of how the activity is being tracked can be found in this article: [Tracking GitHub activity of crypto projects — introducing a better approach](https://medium.com/santiment/tracking-github-activity-of-crypto-projects-introducing-a-better-approach-9fb1af3f1c32)

The difference with the [githubActivity](#github-activity) API is that the actions counted in this API exclude events that are considered non-development related. These are:

* Comments on issues
* Issues created and closed
* Creating of forks
* Comments on commits
* People following an issue

> * Downloading releaes

Reviewer 2: ```suggestion
* Downloading releases
```

* Watching a repository
* Project management events

This metric becomes useful if you compare projects that use the github functionality for project management and issue tracking. They are going to have larger `githubActivity`, compared to projects that use github only for code, as they will generate a lot of events related to issues and tasks.

Parameters:

| Parameter | Description                                                            |
| --------- | ---------------------------------------------------------------------- |
| slug      | The slug of the project                                                |

> | from      | From which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| from      | From which date to return the activity in ISO 8601 format                |
```

> | to        | Till which date to return the volume in ISO 8601 format                |

Reviewer 2: ```suggestion
| to        | Till which date to return the activity in ISO 8601 format                |
```
